### ISSUE TITLE
Did not downloading.

### ISSUE THREAD

**Issue by mrprole** [proninyaroslav/libretorrent#90]
Hello. Just installed app. Trying to download file but it did not downloading. What information i can provide for solving this issue?

**Comment by proninyaroslav** [comment]
Hello. Describe your actions and attach you torrent file (magnet).

**Comment by mrprole** [comment]
OK. I just figured that it did not downloading from http://thepiratebay.org & https://rutracker.org. But there is no problem with https://archive.org. Is there are some type of restriction?

**Comment by proninyaroslav** [comment]
There are no restrictions.

**Comment by mrprole** [comment]
OK. Thats my actions:
![screenshot_20170714-201651](https://user-images.githubusercontent.com/25525606/28222820-b7fd5306-68d1-11e7-8e16-926862f35b7c.png)
![screenshot_20170714-201655](https://user-images.githubusercontent.com/25525606/28222832-c0509cf2-68d1-11e7-9bb6-644c2b4b7bc3.png)
![screenshot_20170714-201659](https://user-images.githubusercontent.com/25525606/28222838-c6ab1b5e-68d1-11e7-89e8-4a3033aaa976.png)
![screenshot_20170714-201706](https://user-images.githubusercontent.com/25525606/28222849-ceeee098-68d1-11e7-8cb2-fa9f250207dc.png)
![screenshot_20170714-201715](https://user-images.githubusercontent.com/25525606/28222861-d52cdb40-68d1-11e7-96c6-a224c72e425e.png)

**Comment by proninyaroslav** [comment]
I understand you are using 1.3 version. Please update to 1.4, this error is fixed in this version.

**Comment by mrprole** [comment]
Just now it downloaded torrent. But i did not updated app. It is still 1.3 version.

**Comment by mrprole** [comment]
Just try to download another torrents that i was not able to download before and it seems there are no problems.

**Comment by proninyaroslav** [comment]
If this problem appears again, let me know.

**Comment by rak5381** [comment]
I might be having this same problem (using the latest commit).

I believe the cause is some combination of trackers and/or ports and/or protocols in the magnet URI.

I opened troublesome magnet URI (including the one above) in a text editor and variously added and removed trackers. Usually, I could remove the trackers one by  a one until the torrent would download properly.

I did this with a few magnet URI until they each worked. Then I tried mixing and matching the trackers that worked from different torrents, and sometimes it would download and others would fail.

I also found that at this particular time for these particular torrents, I could remove all trackers and add only the following to have it work:

tr=udp%3a%2f%2ftracker.leechers-paradise.org%3a6969%2fannounce

In other words, the following magnet downloads just fine:

magnet:?xt=urn:btih:59066769b9ad42da2e508611c33d7c4480b3857b&dn=ubuntu-17.04-desktop-amd64.iso&tr=udp%3a%2f%2ftracker.leechers-paradise.org%3a6969%2fannounce

But sometimes when _adding_ this one to a set of trackers that also worked for that file, and then opening that link in LibreTorrent, the torrent wouldn't work.

I couldn't work out the specific rules involved.

-----

The following is a description of the problem I have for some magnet torrents (without messing around with the URI files):


I noticed a problem with a couple magnet links: when I'd add one, it would appear with the download paused. Pressing the download (or play) button (|>) on the home torrent list page would change the button display to the pause button (||) which _should_ indicate activity. But the text would remain "pause", as in the following image:

![screenshot_20170719-011545](https://user-images.githubusercontent.com/19358528/28352121-a02f55ae-6c22-11e7-8409-485cbaa9aaaa.jpg)

After 30 seconds with the button indicating action, the status text indicating pause, and no connections or downloading, the pause button would return to the play button.Long pressing the torrent to select, or tapping the torrent to enter the settings would also turn the button back to the accurate pause.

"Force announce" did nothing.

Pressing the download (play) button above the tabs in the torrent settings did nothing.

I tried re-downloading the magnet 3 times, with the same effect.

Total active time was listed at 2 seconds, 5 seconds, or 9 seconds even after repeatedly pressing the download button. At least until:

"Force recheck" changed the status to "Checking", and kept the button at  (||) for the duration. Pressing the pause button changed the status listed from "Checking" to "Pause". Then pressing the play button returned the status to "Checking". After 51 minutes, the status remained at "Checking" with no indication any checking was done.

I downloaded the magnet listed above from the same site (TPB), and it had the same problems. The only difference was the trackers.
